_spinnaker/armory_halyard.md

```diff
@@ -211,4 +212,88 @@ hal armory init [parameters]
  * `--path`: The path to where armory-install is already installed
 
 
+## hal armory secrets 
+
+Configure secrets management
+
+#### Usage
+```
+hal armory secrets [subcommands] [parameters]
+```
+
+#### Parameters
+ * `--deployment`: If supplied, use this Halyard deployment. This will _not_ create a new deployment.
+ * `--no-validate`: (*Default*: `false`) Skip validation.
+
+#### Subcommands
+ * `vault`: Configure secrets management with vault
+
 ---
+## hal armory secrets vault
+
+Configure secrets management with vault
+
+#### Usage
+```
+hal armory secrets vault [subcommands] [parameters]
+```
+
+#### Parameters
+ * `--deployment`: If supplied, use this Halyard deployment. This will _not_ create a new deployment.
+ * `--no-validate`: (*Default*: `false`) Skip validation.
+
+#### Subcommands
+ * `disable`: Set secrets as disabled
+ * `edit`: Edit secrets settings
+ * `enable`: Set secrets as enabled
+
+---
+## hal armory secrets vault disable
+
+Set vault secrets as disabled
+
+#### Usage
+```
+hal armory secrets vault disable [parameters]
+```
+
+#### Parameters
+ * `--deployment`: If supplied, use this Halyard deployment. This will _not_ create a new deployment.
+ * `--no-validate`: (*Default*: `false`) Skip validation.
+
+---
+## hal armory secrets vault edit
+
+Edit vault secrets settings
+
+#### Usage
+```
+hal armory secrets vault edit [parameters]
+```
+
+#### Parameters
+ * `--auth-method`: (*Required*) Method used to authenticate with the Vault endpoint. Must be either `KUBERNETES` or `TOKEN`.
+ * `--deployment`: If supplied, use this Halyard deployment. This will _not_ create a new deployment.
+ * `--no-validate`: (*Default*: `false`) Skip validation.
+ * `--path`: Path of the kubernetes authentication backend mount. Default is 'kubernetes'.
```

Comment:
+ "Applies to KUBERNETES authentication method."
Ideally would point to some Hashicorp doc.